sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -185,6 +185,10 @@ case class FileSourceScanExec(
     ret
   }
 
+  private def partitionOnlyAvailableAtRuntime: Boolean = {
+    partitionFilters.exists(ExecSubqueryExpression.hasSubquery)
+  }
```

Comment:
I don't get why this is named as `partitionOnlyAvailableAtRuntime`.

Reply:
actually it should read `partitionsOnlyAvailableAtRuntime` at the suggestion of @cloud-fan 
will rename it soon

Reply:
I need to rethink to link the meaning of `partitionOnlyAvailableAtRuntime` to its implementation and understand its usage below.

I'd prefer to rename this as something like `hasPartitionsAvailableAtRunTime` and add some comments to explain its usage.

Reply:
Ok. Renamed it and added some comments.

Reply:
thanks. LGTM.